src/services/slam/slam.ts

```diff
@@ -0,0 +1,9 @@
+import type { Resource } from '../../types';
+import type { GetPositionResponse, Timestamp } from './types';
+
+export interface Slam extends Resource {
+  getPosition: () => Promise<GetPositionResponse>;
+  getPointCloudMap: () => Promise<string | Uint8Array>;
+  getInternalState: () => Promise<string | Uint8Array>;
```

Comment:
there should be a method on the protobuf response object like `response.get<FIELD_NAME>_asU8` (see camera component client for an example), that returns the response data as a `Uint8Array`. The more we can constrain the return type the better.

```suggestion
  getPointCloudMap: () => Promise<Uint8Array>;
  getInternalState: () => Promise<Uint8Array>;
```

Reply:
This sounds great. I was also hesitant about this at first but thought it would come up in the review (which it did!)